test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs

```diff
@@ -0,0 +1,360 @@
+            _userSdkBaseDir = Path.Combine(_userDir, ".dotnet", RuntimeEnvironment.RuntimeArchitecture, "sdk");
+            _exeSdkBaseDir = Path.Combine(_executableDir, "sdk");
+
+            // Create directories
+            Directory.CreateDirectory(_cwdSdkBaseDir);
+            Directory.CreateDirectory(_userSdkBaseDir);
+            Directory.CreateDirectory(_exeSdkBaseDir);
+
+            // Restore and build PortableApp from exe dir
+            RepoDirectories = new RepoDirectoriesProvider(builtDotnet: _executableDir);
+            PreviouslyBuiltAndRestoredPortableTestProjectFixture = new TestProjectFixture("PortableApp", RepoDirectories)
+                .EnsureRestored(RepoDirectories.CorehostPackages, RepoDirectories.CorehostDummyPackages)
+                .BuildProject();
+            var fixture = PreviouslyBuiltAndRestoredPortableTestProjectFixture;
+
+            // Set a dummy framework version (9999.0.0) in the cwd sharedFx location. We will
+            // always pick the framework from cwd to avoid interference with the sharedFxLookup
+            // test folders in the user dir
+            string cwdDummyFxVersion = Path.Combine(_currentWorkingDir, "shared", "Microsoft.NETCore.App", "9999.0.0");
+            string builtSharedFxDir = fixture.BuiltDotnet.GreatestVersionSharedFxPath;
+            CopyDirectory(builtSharedFxDir, cwdDummyFxVersion);
+
+            // The actual SDK version can be obtained from the built fixture. We'll use it to
+            // locate the sdkDir from which we can get the files contained in the version folder
+            string sdkBaseDir = Path.Combine(fixture.SdkDotnet.BinPath, "sdk");
+
+            var sdkVersionDirs = Directory.EnumerateDirectories(sdkBaseDir);
+            string greatestVersionSdk = sdkVersionDirs
+                .OrderByDescending(p => p.ToLower())
+                .First();
+
+            _sdkDir = Path.Combine(sdkBaseDir, greatestVersionSdk);
+
+            // Trace messages used to identify from which folder the SDK was picked
+            _cwdSelectedMessage = $"Using dotnet SDK dll=[{_cwdSdkBaseDir}";
+            _userSelectedMessage = $"Using dotnet SDK dll=[{_userSdkBaseDir}";
+            _exeSelectedMessage = $"Using dotnet SDK dll=[{_exeSdkBaseDir}";
+        }
+
+        [Fact]
+        public void SdkLookup_Must_Verify_Folders_in_the_Correct_Order()
+        {
+            var fixture = PreviouslyBuiltAndRestoredPortableTestProjectFixture
+                .Copy();
+
+            var dotnet = fixture.BuiltDotnet;
+
+            // Add a dummy version in the exe dir
+            AddAvailableSdkVersions(_exeSdkBaseDir, "9999.0.0-dummy");
+
+            // Specified CLI version: none
+            // CWD: empty
+            // User: empty
+            // Exe: 9999.0.0
+            // Expected: 9999.0.0 from exe dir
+            dotnet.Exec("help")
+                .WorkingDirectory(_currentWorkingDir)
```

Comment:
@gkhanna79 I don't think so. The only possible negative test would be if all three folders were empty. That's troublesome to test because any version from the sdk in the global location would be picked, so the test might fail in some machines.